src/Controls/tests/UITests/Tests/_ViewUITests.cs

```diff
@@ -51,6 +52,32 @@ public virtual void _IsVisible()
 
 			Assert.AreEqual(0, viewPost.Count);
 		}
-	}
 
+		internal StateViewContainerRemote GoToStateRemote([CallerMemberName] string? testName = null)
+		{
+			_ = testName ?? throw new ArgumentNullException(nameof(testName));
+
+			var remote = new StateViewContainerRemote(UITestContext, testName);
+			remote.GoTo();
+			return remote;
+		}
+
+		internal EventViewContainerRemote GoToEventRemote([CallerMemberName] string? testName = null)
+		{
+			_ = testName ?? throw new ArgumentNullException(nameof(testName));
+
+			var remote = new EventViewContainerRemote(UITestContext, testName);
+			remote.GoTo();
+			return remote;
+		}
+
+		internal ViewContainerRemote GoToRemote([CallerMemberName] string? testName = null)
+		{
+			_ = testName ?? throw new ArgumentNullException(nameof(testName));
+
+			var remote = new ViewContainerRemote(UITestContext, testName);
+			remote.GoTo();
```

Comment:
Could we wait for any element ot at least some ms?
![image](https://github.com/dotnet/maui/assets/6755973/7b1e58cf-be53-4942-9b86-9eeda23db732)
It seems that the snapshot is taken earlier than expected.

Reply:
I just did a bad thing, should be working hopefully

Reply:
Added a wait as well, since time moves so fast on CI or something...

Reply:
Added a wait as well, since time moves so fast on CI or something...